topics/ui/principles/empty_state.md

```diff
@@ -10,144 +10,261 @@
 
 </tldr>
 
-Provide instructions in an empty UI area on how to fill it with data. Empty UI areas described here are:
-
-* [Tool window, list, tree or table](#tool-windows-lists-trees-and-tables)
-* [Details area in a master-detail layout](#master-detail-layout)
-* Empty search results — TBD
-
-## Tool windows, lists, trees and tables
-
-Instructional text for these UI areas consists of three parts:
-
-![](database-tw-callouts.png){width=626}
-
-### 1. Reason why empty
-
-Explain the current state.
-
-The default pattern is "_No [entity] added._" If _added, created, configured_ or other such verbs act as synonyms in a particular case, use the verb _added_ for consistency.
-
-<table>
-    <tr>
-        <td width="50%"><format color="Red" style="bold">Incorrect</format></td>
-        <td width="50%"><format color="Green" style="bold"> Correct</format></td>
-    </tr>
-    <tr>
-        <td><img src="libraries-before.png" alt="" width="241" /> Avoid &quot;Nothing to show&quot; as it is not informative.</td>
-        <td><img src="libraries-after.png" alt="" width="241" /></td>
-    </tr>
+Empty states of tool windows, lists, trees, tables, master-detail layouts, or other container components should be informative for the users.
+Fill in the empty states with the following:
+- [A reason why the container is empty](#reason-why-empty)
+- [Actions to fill in the area](#actions-to-fill-the-area)
+- [A link to the corresponding help topic](#help-topic)
+
+<img src="empty_state_structure.png" alt="Empty state of a tool window" width="706"/>
+
+## Reason why empty
+
+### Explain the current state
+
+The default pattern is `No [entity] added.` If `added`, `created`, `configured`, or other words act as synonyms in a particular case, use `added` for consistency.
+
+<table style="none" border="false" column-width="fixed">
+  <tr>
+    <td>
+      <format color="369650" style="bold">Correct</format>
+      <img src="empty_state_clear_reason.png" alt="Correct text in the empty state" />
```

Comment:
Should these pics be about the library not the SQL dialects? Otherwise, we have the same illustration in nearby sections